cider/exec/module/batch/CiderBatch.cpp

```diff
@@ -352,6 +354,22 @@ const void** CiderBatch::getChildrenArrayPtr() const {
   return const_cast<const void**>(reinterpret_cast<void**>(arrow_array_->children));
 }
 
+const void* CiderBatch::arrow_column(int32_t col_id) const {
+  CHECK(!isMoved());
+  const void* buf = arrow_array_->children[col_id]->buffers[1];
+  return buf;
+}
+
+void* CiderBatch::getDataBuffersPtr() const {
+  CHECK(!isMoved());
+  return const_cast<void*>(arrow_array_->buffers[1]);
+}
+
+const void* CiderBatch::getNullBuffersPtr() const {
+  CHECK(!isMoved());
+  return arrow_array_->buffers[0];
+}
+
```

Comment:
Seems `getDataBuffersPtr()` and `getNullBuffersPtr()` are equal to `getBuffersPtr()[0]`/`getBuffersPtr()[0]`. Should we use the `getBuffersPtr()` instead? I mean seems there is a lot API defined in CiderBatch and we should be careful when adding new ones.

Reply:
I think ```getBuffersPtr()[0]``` and ```getBuffersPtr()[1]``` are not readable. I considered them as replacement for original ciderbatch api :```get_null_vecs()``` and ```table()```

Reply:
The problem is that data buffer is not always `arrow_array_->buffers[1]`, take `varchar` for example, it includes `buffers[1]` and `buffers[2]`.  We don't need to consider readiness since we all need to understand the internal arrow format.